pip_audit/_dependency_source/requirement.py

```diff
@@ -164,15 +138,46 @@ def _fix_file(self, filename: Path, fix_version: ResolvedFixVersion) -> None:
 
         # Now write out the new requirements file
         with filename.open("w") as f:
+            fixed = False
             for req in req_list:
                 if (
                     req.name == fix_version.dep.name
                     and req.specifier.contains(fix_version.dep.version)
                     and not req.specifier.contains(fix_version.version)
                 ):
                     req.specifier = SpecifierSet(f"=={fix_version.version}")
+                    fixed = True
                 assert req.marker is None or req.marker.evaluate()
-                f.write(str(req) + os.linesep)
+                print(str(req), file=f)
+
+            # The vulnerable dependency may not be explicitly listed in the requirements file if it
+            # is a subdependency of a requirement. In this case, we should explicitly add the fixed
+            # dependency into the requirements file.
+            #
+            # To know whether this is the case, we'll need to resolve dependencies if we haven't
+            # already in order to figure out whether this subdependency belongs to this file or
+            # another.
+            try:
+                if not fixed:
+                    origin_reqs: Set[Requirement] = set()
+                    for req, dep in self._collect_cached_deps(filename, list(reqs.values())):
+                        if fix_version.dep == dep:
+                            origin_reqs.add(req)
+                    if origin_reqs:
+                        logger.warning(
+                            "added fixed subdependency explicitly to requirements file "
+                            f"{filename}: {fix_version.dep.canonical_name}"
+                        )
+                        origin_reqs_formatted = ",".join(
+                            [str(req) for req in sorted(list(origin_reqs), key=lambda x: x.name)]
+                        )
+                        print(
+                            f"# pip-audit: subdependency fixed via {origin_reqs_formatted}",
```

Comment:
Indent, to mimic what `pip-compile` does:

```suggestion
                            f"    # pip-audit: subdependency fixed via {origin_reqs_formatted}",
```